mllib-local/src/main/scala/org/apache/spark/ml/linalg/Vectors.scala

```diff
@@ -109,6 +109,16 @@ sealed trait Vector extends Serializable {
     throw new UnsupportedOperationException(s"copy is not implemented for ${this.getClass}.")
   }
 
+  /**
+   * Applies a function `f` to all the elements of dense and sparse vector.
+   *
+   * @param f the function takes two parameters where the first parameter is the index of
+   *          the vector with type `Int`, and the second parameter is the corresponding value
+   *          with type `Double`.
+   */
+  private[spark] def foreach(f: (Int, Double) => Unit): Unit =
```

Comment:
I don't mind it, but do you need foreach and foreachNonZero if you expose iterators internally? 
Also I wonder if there is a perf overhead to implementing this as an iterator. Probably not enough to care about.

Reply:
Seem not needed, I will remove `foreach` and `foreachNonZero`

Reply:
emmm, for convenience, I tend to keep foreach and foreachNonZero

Reply:
OK, up to you